cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/co/cask/cdap/etl/common/ExternalDatasets.java

```diff
@@ -79,6 +79,9 @@ public static Input makeTrackable(Admin admin, Input input) {
           throw new IllegalArgumentException(
             "An external source cannot have the same name as an existing CDAP Dataset instance " + inputName);
         }
+        // if the dataset already exists with the same type then it must have been created for tracking during the
+        // deploy stage. Just update the properties in this case.
+        admin.updateDataset(inputName, DatasetProperties.of(arguments));
```

Comment:
should not do this. We don't update datasets anywhere else and this can easily lead to flip flopping properties.